File: designPatterns/Behavioral_patterns/State/python/State.py
from abc import ABC, abstractmethod


class State(ABC):
    """
    The base State class declares methods that all Concrete State should
    implement and also provides a backreference to the Context object,
    associated with the State. This backreference can be used by States to
    transition the Context to another State.
    """

    @property
    def context(self):
        return self._context

    @context.setter
    def context(self, context):
        self._context = context

    @abstractmethod
    def handle1(self) -> None:
        pass

    @abstractmethod
    def handle2(self) -> None:
        pass

File: designPatterns/Behavioral_patterns/State/javascript/State.js
/**
 * The base State class declares methods that all Concrete State should
 * implement and also provides a backreference to the Context object, associated
 * with the State. This backreference can be used by States to transition the
 * Context to another State.
 */
class State {
  setContext(context) {
    this.context = context;
  }

  handle1() {}

  handle2() {}
}

module.exports = State;
